neurom/view/view.py

```diff
@@ -281,80 +282,77 @@ def plot_neuron3d(ax, nrn, neurite_type=NeuriteType.all,
     ax.set_title(nrn.name)
 
 
-def _generate_collection(group, ax, ctype, colors):
-    '''Render rectangle collection'''
-    color = TREE_COLOR[ctype]
-
-    # generate segment collection
-    collection = PolyCollection(group, closed=False, antialiaseds=True,
-                                edgecolors='face', facecolors=color)
-
-    # add it to the axes
-    ax.add_collection(collection)
-
-    # dummy plot for the legend
-    if color not in colors:
-        label = str(ctype).replace('NeuriteType.', '').replace('_', ' ').capitalize()
-        ax.plot((0., 0.), (0., 0.), c=color, label=label)
-        colors.add(color)
+def _get_dendrogram_legend(dendrogram):
+    '''Generates labels legend for dendrogram.
 
+    Because dendrogram is rendered as patches, we need to manually label it.
+    Args:
+        dendrogram (Dendrogram): dendrogram
 
-def _render_dendrogram(dnd, ax, displacement):
-    '''Renders dendrogram'''
-    # set of unique colors that reflect the set of types of the neurites
-    colors = set()
+    Returns:
+        List of legend handles.
+    '''
+    def neurite_legend(neurite_type):
+        return Line2D([0], [0], color=TREE_COLOR[neurite_type], lw=2, label=neurite_type.name)
 
-    for n, (indices, ctype) in enumerate(zip(dnd.groups, dnd.types)):
+    if dendrogram.neurite_type == NeuriteType.soma:
+        handles = {d.neurite_type: neurite_legend(d.neurite_type)
+                   for d in [dendrogram] + dendrogram.children}
+        return handles.values()
+    return [neurite_legend(dendrogram.neurite_type)]
 
-        # slice rectangles array for the current neurite
-        group = dnd.data[indices[0]:indices[1]]
 
-        if n > 0:
-            # displace the neurites by half of their maximum x dimension
-            # plus half of the previous neurite's maxmimum x dimension
-            displacement += 0.5 * (dnd.dims[n - 1][0] + dnd.dims[n][0])
+def _as_dendrogram_polygon(coords, color):
+    return Polygon(coords, color=color, fill=True)
 
-        # arrange the trees without overlapping with each other
-        group += (displacement, 0.)
 
-        # create the polygonal collection of the dendrogram
-        # segments
-        _generate_collection(group, ax, ctype, colors)
+def _as_dendrogram_line(start, end, color):
+    return FancyArrowPatch(start, end, arrowstyle='-', color=color, lw=2, shrinkA=0, shrinkB=0)
 
-    soma_square = dnd.soma
 
-    if soma_square is not None:
+def _get_dendrogram_shapes(dendrogram, positions, show_diameter):
```

Comment:
I assume this is here instead of in `_dendrogram.py` so that `_dendrogram.py` doesn't have matplotlib dependencies?

does that mean `_dendrogram.py` should be 'public', so you can reuse the tree building mechanism for other software?

Reply:
Yes. I intend to make it public.